coalib/settings/Setting.py

```diff
@@ -163,7 +163,11 @@ def __path__(self, origin=None, glob_escape_origin=False):
         :raises ValueError:        If no origin is specified in the setting
                                    nor the given origin parameter.
         """
-        strrep = str(self).strip()
+        if hasattr(self, 'value'):
+            strrep = self.value.strip()
```

Comment:
`str` objects are used as `Section` objects? WTF?!

>I can replace `if hasattr(self, 'value'):` with `if isinstance(self, Setting):`

That should work yeah. But the fact that we need it is baffling!